docs/design/process_group_crd.md

```diff
@@ -0,0 +1,103 @@
+Additional to the fields above that are taken from the `FDB cluster spec` we will add the fields from [ProcessGroupStatus](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/master/docs/cluster_spec.md#processgroupstatus) to the Spec:
+
+- `processClass`
+- `remove`
+- `excluded`
+- `exclusionSkipped`
+
+`addresses` is already included in the status field and the `processGroupID` should be used as the name of the `FDBProcessGroup`.
+Currently we have the following `ProcessGroupConditions` as part of the `ProcessGroupStatus`:
+
+```go
+const (
+	// NotConnecting represents a process group that doesn't connect to the cluster.
+	NotConnecting ProcessGroupConditionType = "NotConnecting"
+	// IncorrectPodSpec represents a process group that has an incorrect Pod spec.
+	IncorrectPodSpec ProcessGroupConditionType = "IncorrectPodSpec"
+	// IncorrectConfigMap represents a process group that has an incorrect ConfigMap.
+	IncorrectConfigMap ProcessGroupConditionType = "IncorrectConfigMap"
+	// IncorrectCommandLine represents a process group that has an incorrect commandline configuration.
+	IncorrectCommandLine ProcessGroupConditionType = "IncorrectCommandLine"
+	// PodFailing represents a process group which Pod keeps failing.
+	PodFailing ProcessGroupConditionType = "PodFailing"
+	// MissingPod represents a process group that doesn't have a Pod assigned.
+	MissingPod ProcessGroupConditionType = "MissingPod"
+	// MissingPVC represents a process group that doesn't have a PVC assigned.
+	MissingPVC ProcessGroupConditionType = "MissingPVC"
+	// MissingService represents a process group that doesn't have a Service assigned.
+	MissingService ProcessGroupConditionType = "MissingService"
+	// MissingProcesses represents a process group that misses a process.
+	MissingProcesses ProcessGroupConditionType = "MissingProcesses"
+)
+```
+
+Some of them are currently unused and most of them can simply be replaced by the `FDBProcessGroup` status.
+During the reconcilation the `FDB cluster controller` will only evaluate the state of the `FDBProcessGroup` and compare it with the current database state.
+Each `FDBProcessGroup` will have the `fdb-process-class` and `fdb-cluster-name`  to make if simpler to filter only for the relevant `FDBProcessGroup`.
+Instead of using the `fdb-instance-id` label we will name the `FDBProcessGroup` accordingly.
+
+This change has the benefit that we can separate some of the tasks handled by the controller and prevent some conflicts:
+
+- The `FDB cluster controller` will read-write the `FDBProcessGroup.Spec` and read-only the `FDBProcessGroup.Status`.
+- The `FDB ProcessGroup controller` will read-only `FDBProcessGroup.Spec` and read-write the `FDBProcessGroup.Spec`.
+
+In this case we prevent conflicts since each controller takes care of either the `Spec` or the `Status` but not both.
+This will be major change and will require some refactoring at multiple places.
```

Comment:
TODO: write a short paragraph that the cluster controller will be responsible for exclude/include/remove etc. how it will interact with the `ProcessGroup`